opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/zuul/TracePreZuulFilter.java

```diff
@@ -0,0 +1,55 @@
+package io.opentracing.contrib.spring.cloud.zuul;
+
+import com.netflix.zuul.ZuulFilter;
+import com.netflix.zuul.context.RequestContext;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.propagation.Format;
+import io.opentracing.propagation.TextMapInjectAdapter;
+import io.opentracing.tag.Tags;
+
+class TracePreZuulFilter extends ZuulFilter {
+    static final String COMPONENT_NAME = "zuul";
+
+    private final Tracer tracer;
+
+    TracePreZuulFilter(Tracer tracer) {
+        this.tracer = tracer;
+    }
+
+    @Override
+    public String filterType() {
+        return "pre";
```

Comment:
isn't this defined as a constant somewhere in zuul?

Reply:
yes, but in newer version than ours

Reply:
:/ TODO would work also :)